check-pr-status.mjs

```diff
@@ -7,22 +7,23 @@ import { setTimeout } from 'timers/promises';
 const tries = 10;
 const retryDelay = 30000;
 
-await setTimeout(retryDelay);
+export default async function(github, owner, repository, pull_number) {
+  const repo = repository.replace(new RegExp(`${owner}/`), '')
+  await setTimeout(retryDelay);
 
-for (let t = 0; t < tries; t++) {
-  try {
-    const [repo, pull_number] = process.argv.slice(2);
+  for (let t = 0; t < tries; t++) {
+    try {
+      const { data } = await github.rest.pulls.get({owner, repo, pull_number})
 
-    const data = await (await fetch(`https://api.github.com/repos/${repo}/pulls/${pull_number}`)).json();
-
-    console.log(data);
-    if (data.mergeable_state === 'clean') {
-      process.exit(0);
+      console.log(data);
+      if (data.mergeable_state === 'clean') {
```

Comment:
I wasn't really able to test this one on my branch as I had issues getting the test action to run for some reason... How did you test?

Reply:
Created a new repo with some simple js scripts and a small test suite, kept polling the GitHub API until the mergeable_state became 'clean'. At creation, it's set to 'unknown'. Then it goes 'unstable' until checks pass, which set that variable to 'clean'

Reply:
Repo was public, of course

Reply:
Right, but it's never clean due to missing approval, and if approval is added, it might be mergable right away. I _think_.

Anyways, if automerge doesn't work, we can iterate. Getting the PR to begin with is a great start as it can be merged from mobile, no need to run the script from desktop